src/mapping_navigation/scripts/TestGUI.py

```diff
@@ -360,58 +290,116 @@
+
+        points = []
+        for j in range(num_points):
+            x = start_point.x + j * interval * (end_point.x - start_point.x) / distance
+            y = start_point.y + j * interval * (end_point.y - start_point.y) / distance
+            points.append(Point(x, y, RoadSegmentType.STRAIGHT))
+        points.append(end_point)
+        new_path.append(points)
+
+    print(new_path)
+    merged_list = []
+    for sublist in new_path:
+        merged_list.extend(sublist)
+
+    return merged_list
 
 # create path from clicking on canvas
 def handle_canvas_m1(event):
-    global map_model, current_image
+    global map_model, current_image, dest_node
     x, y = event.x, event.y
 
     #delete line and path if already created
     canvas.delete("line")
-    if not(map_model.empty()):
+    if not map_model.empty():
         map_model.delete_path(0)
 
-    #choosing node to travel too
-    point_found = False
-    endpoint = -1
-    if current_image == CITY:
-        for point in C_pointList:
-            if abs(x - point.x) < 10 and abs(y - point.y) < 10:
-                endpoint = point
-                point_found = True
-                break
-    else:
-        for point in NH_pointList:
-            if abs(x - point.x) < 10 and abs(y - point.y) < 10:
-                endpoint = point
-                point_found = True
-                break
-
-    if point_found:
-        if current_image == CITY:
-            path = find_shortest_path(C_G, C_startpoint, endpoint)
-        else:
-            path = find_shortest_path(NH_G, NH_startpoint, endpoint)
-    else:
-        #if no node is clicked
-        return
+    #delete destination node from graph when creating new path
+    if not dest_node.x is None:
+        if current_image == NH and NH_G.has_node(dest_node):
+            neighbor_nodes = list(NH_G.neighbors(dest_node))
+            NH_G.remove_node(dest_node)
+            if len(neighbor_nodes) > 1:
+                node1, node2 = neighbor_nodes
+                NH_G.add_edge(node1, node2, weight=find_edge_weight(node1, node2))
+        elif current_image == CITY and C_G.has_node(dest_node):
+            neighbor_nodes = list(C_G.neighbors(dest_node))
+            C_G.remove_node(dest_node)
+            if len(neighbor_nodes) > 1:
+                node1, node2 = neighbor_nodes
+                C_G.add_edge(node1, node2, weight=find_edge_weight(node1, node2))
+
+    #adding chosen node to graph
+    dest_node = Point(x, y, RoadSegmentType.STRAIGHT)
+    path: List[Point] = []
+    if current_image == NH:
+        add_dest_node(NH_G, dest_node, NH_pointList)
+        path = find_shortest_path(NH_G, NH_startpoint, dest_node)
+    elif current_image == CITY:
+        add_dest_node(C_G, dest_node, C_pointList)
+        path = find_shortest_path(C_G, C_startpoint, dest_node)
 
     #create line on canvas showing generated path
-    draw_line(path)
+    better_path = optimize_path(path)
+    draw_line(better_path)
+    print(path)
```

Comment:
This line is cluttering the terminal when we run the tool

Reply:
Yeah it was used for testing purposes, will delete all the print statements in my next PR once i finished implementing the curve